lib/clAdapters/jira/index.js

```diff
@@ -56,11 +61,61 @@ export default class JiraAdapter extends Adapter {
     });
   }
 
-  async getIssueHierarchy() {
-    return await this.makeRequest('issue/createmeta');
+  async getAllIssues(params) {
+    const requestParams = params || {};
+    requestParams.startAt = 0;
+    requestParams.maxResults = 50;
+
+    let resultCount,
+        issues = [];
+    do {
+      const result = await this.makeRequest('search', requestParams),
+            data = JSON.parse(result.data);
+
+      if (data.issues && data.issues.length) {
+        resultCount = data.issues.length;
+        issues = issues.concat(data.issues);
+        requestParams.startAt += resultCount;
+      } else {
+        resultCount = 0;
+      }
+    } while (resultCount && resultCount === requestParams.maxResults);
+
+    return issues;
   }
 
   async runConnectionTest() {
     return await this.makeRequest('myself');
   }
+
+  async getIssueHierarchy() {
+    return await this.makeRequest('issue/createmeta');
```

Comment:
minor nitpick, but for functions like this when you are immediately awaiting a promise, you might as well just write it as 

```javascript
getIssueHierarchy() {
  return this.makeRequest('issue/createmeta');
}
```

Which is has the same result, but avoids the overhead of transpiling the async function to a generator function, and calling the generator function